### ISSUE TITLE
git: `post_install` command breaks behaviour and creates dark pattern

### ISSUE THREAD

**Issue by FriendlyNeighborhoodShane** [ScoopInstaller/Main#2190]
The `git` manifest has a `post_install` command (last commit de25ea9d9f1a63ea416a98630f32785a789e9810 by @amaechler) that sets/overwrites the user configuration for the credential manager. It should not be there for several reasons:

First of all, it is incredibly surprising and just plain _wrong_ behavior for a package manager, or anybody else, to touch a user configuration. Let alone on every update just plain _overwrite_ the user choice. On every update I choose my preferred credential helper and the next update overwrites it. The commit justifies it by linking another commit in the git repository, but you will notice that that commit changes the _default_. It does not change anything in the user config (rightfully, as the user config is not part of the package. It is part of the user's environment) but if you have selected a different manager, it respects your choice and doesn't force it on you. (Speaking of which, why was the need for that line even felt, seeing as the default was literally changed upstream?)

Secondly and more importantly, it just trashes the upstream's sane behavior. The upstream git-for-windows comes with a `credential-helper-selector` GUI program set to the default helper. The first time that you push, it is supposed to prompt you to choose the credential helper you want, and it sets your choice in the 'system' config (See? They know not to touch the user config). The problem is, the user config takes rightful priority over the system config, and so the option forced on you by the manifest command takes priority. That command is not expecting to be executed in this way, and you get a loop where the GUI program is launched multiple times per push to ask you but none of the choices you make change anything. Upstream comes with a handy GUI to help newcomers decide their credential helper and because of this command both that behavior and any choice an already-familiar user might have made earlier are ignored.

Worst of all, this leads to a dark pattern. Even if the user, at every point they are asked, having selected any other helper or none, this will force the usage of the `manager-core` helper. Despite github phrasing out passwords, I'm sure many still use it, not to mention other git hosters. Many naive users may not even realize that their passwords are being stored in plaintext, accessible to anybody with an unprivileged shell command, without so much as the OS keychain protecting it.

It's as simple as:
```
printf '%s\n' protocol=https host=github.com | git credential-manager-core get
```

TLDR: I see no need for that command whatsoever. git-for-windows on a fresh install already comes with a GUI to set up your credential helper, and not only does this break that sane behavior and deliberately overwrites user choice for no reason, it opens up a security hole to naive users.

**Comment by FriendlyNeighborhoodShane** [comment]
Browsing through the history; It was this commit d067f7bf8c799e16fdbacc342589df347396fc57 by @Ash258 that introduced the `post_install` command to the manifest at first. I believe it should be removed now, because as I explained above it has literally no use and causes all sorts of problems.

**Comment by FriendlyNeighborhoodShane** [comment]
By the way, if anybody reading this has realized that they never wanted to use a credential manager, or use a better manager than this terrible one, or use a better authentication method like SSH keys, but this dark pattern might have lead to your credentials being stored on your computers:
```
printf '%s\n' protocol=https host=github.com | git credential-manager-core erase
```
With the appropriate `host` URL.

**Comment by dwettstein** [comment]
Duplicate of #1617

**Comment by r15ch13** [comment]
Fixed

**Comment by FriendlyNeighborhoodShane** [comment]
@r15ch13 Could you also please take a look at #2185 ? And if you are also a collaborator for https://github.com/lukesampson/scoop, https://github.com/lukesampson/scoop/pull/4359?

**Comment by pratikpc** [comment]
@FriendlyNeighborhoodShane if you notice, that PR by @Ash258 only formats the file.

The credential helper was certainly added before that.
